hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/KeyGenUtils.java

```diff
@@ -131,6 +131,9 @@ public static String getRecordPartitionPath(GenericRecord record, List<String> p
       } else {
         if (encodePartitionPath) {
           fieldVal = PartitionPathEncodeUtils.escapePathName(fieldVal);
+        } else {
+          // Hive doesn't respect the space at the end, so remove it to avoid duplicate keys error
+          fieldVal = fieldVal.trim();
```

Comment:
I'm not sure if I follow the fix. So, if the value is trimmed then isn't it same as untrimmed value and hence duplicate partition key?

Reply:
Yeah, the fix changes the partition value which may not be what the user desires.

Reply:
In my case, the user's data has some wrong partition values with space at the end(space shouldn't exist), then the data has been separated into two HDFS paths, and those two paths cannot be synced to Hive.
I did this fix to avoid upstream change and hive sync errors. 
if the user really wants two hdfs paths and two partitions in hive, they should encode partition path

Reply:
Hudi will faithfully write the records being passed without these side-effects to user data. This is not the right place to correct incorrect input data.  It should be handled by the caller.